itest/multi_send_test.go

```diff
@@ -0,0 +1,289 @@
+		anchorInternalKeyDesc1, asset.V0, chainParams,
+	)
+	fundRespSecondBatch := fundPacket(t, aliceTapd, vPktSecondBatch)
+	tranche3, err := tappsbt.Decode(fundRespSecondBatch.FundedPsbt)
+	require.NoError(t.t, err)
+
+	activePackets = append(activePackets, tranche3)
+
+	// We now need to update the outputs of the passive packets to also go
+	// to Bob.
+	scriptKeys := []asset.ScriptKey{scriptKey1}
+	for idx, activePacket := range activePackets {
+		if idx > 0 {
+			scriptKey, _ := DeriveKeys(t.t, bobTapd)
+			activePacket.Outputs[0].ScriptKey = scriptKey
+			activePacket.Outputs[0].Asset.ScriptKey = scriptKey
+			scriptKeys = append(scriptKeys, scriptKey)
+		}
+
+		activePacket.Outputs[0].AnchorOutputBip32Derivation = nil
+		activePacket.Outputs[0].AnchorOutputTaprootBip32Derivation = nil
+		activePacket.Outputs[0].SetAnchorInternalKey(
+			anchorInternalKeyDesc1, chainParams.HDCoinType,
+		)
+		activePacket.Outputs[0].AnchorOutputIndex = 0
+	}
+
+	// Let's now sign all the active packets.
+	signedPackets := make([][]byte, len(activePackets))
+	for idx := range activePackets {
+		signedPacket := signVirtualPacket(
+			t.t, aliceTapd, activePackets[idx],
+		)
+
+		signedPackets[idx], err = tappsbt.Encode(signedPacket)
+		require.NoError(t.t, err)
+	}
+
+	// Now we'll attempt to complete the transfer.
+	sendResp, err := aliceTapd.AnchorVirtualPsbts(
+		ctxt, &wrpc.AnchorVirtualPsbtsRequest{
+			VirtualPsbts: signedPackets,
+		},
+	)
+	require.NoError(t.t, err)
+
+	t.Logf("Send response: %v", toJSON(t.t, sendResp))
+
+	ConfirmAndAssertOutboundTransferWithOutputs(
+		t.t, t.lndHarness.Miner.Client, aliceTapd, sendResp,
+		assetXTranche1ID[:], []uint64{300, 300, 300}, 0, 1, 3,
+	)
+
+	// This is an interactive transfer, so we do need to manually send the
+	// proofs from the sender to the receiver.
+	_ = sendProof(
+		t, aliceTapd, bobTapd, sendResp,
+		scriptKeys[0].PubKey.SerializeCompressed(),
+		assetXTranche1.AssetGenesis,
+	)
+	_ = sendProof(
+		t, aliceTapd, bobTapd, sendResp,
+		scriptKeys[1].PubKey.SerializeCompressed(),
+		assetXTranche2.AssetGenesis,
+	)
+	_ = sendProof(
+		t, aliceTapd, bobTapd, sendResp,
+		scriptKeys[2].PubKey.SerializeCompressed(),
+		assetXTranche3.AssetGenesis,
+	)
+
+	aliceAssets, err := aliceTapd.ListAssets(
+		ctxt, &taprpc.ListAssetRequest{},
+	)
+	require.NoError(t.t, err)
+	require.Len(t.t, aliceAssets.Assets, 5)
+
+	t.Logf("Alice assets: %v", toJSON(t.t, aliceAssets))
+
+	bobAssets, err := bobTapd.ListAssets(ctxt, &taprpc.ListAssetRequest{})
+	require.NoError(t.t, err)
+	require.Len(t.t, bobAssets.Assets, 3)
+
+	t.Logf("Bob assets: %v", toJSON(t.t, bobAssets))
+
+	// Next, we make sure we can still send out the passive assets.
+	sendAssetAndAssert(
+		ctxt, t, t.tapd, secondTapd, assetP.Amount, 0,
+		assetP.AssetGenesis, assetP, 1, 2, 1,
+	)
+	sendAssetAndAssert(
+		ctxt, t, t.tapd, secondTapd, assetQ.Amount, 0,
+		assetQ.AssetGenesis, assetQ, 2, 3, 2,
+	)
```

Comment:
should we also be doing some checks for the assetY? it's not clear why we minted it together with assetX in the same batches or what its purpose was

Reply:
Correct as well. I initially wanted to add another test case with tombstone outputs, and those two assets (P and Q) were just added for that. So I added that additional case now.